src/components/progress-bar.tsx

```diff
@@ -27,26 +25,39 @@ const ProgressBar = ( {
 
 export default ProgressBar;
 
+type ProgressBarWithAutoIncrementProps = {
+	maxValue: number;
+	value: number;
+	increment?: number;
+};
+
 export function ProgressBarWithAutoIncrement( {
 	maxValue,
-	initialValue,
+	value,
 	increment = 2,
-}: {
-	maxValue: number;
-	initialValue: number;
-	increment?: number;
-} ) {
-	const [ value, setValue ] = useState( initialValue );
+}: ProgressBarWithAutoIncrementProps ) {
+	const [ animatedValue, setAnimatedValue ] = useState( value );
 
 	useEffect( () => {
+		if ( value > animatedValue ) {
+			setAnimatedValue( value );
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ value ] );
+
+	useEffect( () => {
+		const maximumAutoIncrementValue = maxValue * 0.95;
+
 		const interval = setInterval( () => {
```

Comment:
Maybe I don't understand how it works, but it seems it should be setTimeout, since we run this effect every time when we update `animatedValue`.

Reply:
Good call 👍